### PR TITLE
fix: update Network page for networks with crawl not found

### PR DESCRIPTION
**Pull request by pdp2121**
## High Level Overview of Change

Not all rippled/ entrypoint can be crawled.  In this case we should show users an error message on the Network pages for custom networks.

This issue is present with the hooks testnet and iou escrows testnet.  

[XRPL Hooks Amendment](http://hooks-testnet-v2.xrpl-labs.com/) 
[icv2.ws.transia.co:443](http://icv2.ws.transia.co:443/)

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [x] Updated files to React Hooks
- [x] Updated files to TypeScript

## Before / After

After the change, the Network tab for networks that cannot be crawls should be as below:

Screenshot 2023-01-31 at 4 36 03 PM

### CODE REVIEW

File: src/containers/Network/index.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+interface NetworkProps {
+  match: {
+    path: string
+    params: {
+      identifier: string
+      tab: string
+    }
+  }
+}
+const Network = (props: NetworkProps) => {
```
Comment (ckniffen) [inline src/containers/Network/index.tsx:20]: Use named export.

File: src/containers/Network/index.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const { t } = useTranslation()
+  const { match } = props
```
Comment (ckniffen) [inline src/containers/Network/index.tsx:22]: Leverage `useMatch` instead.

File: src/containers/App/App.jsx [hunk 3]
```diff
@@ -55,7 +55,11 @@ const App = (props) => {
       axios
         .get(`${process.env.REACT_APP_DATA_URL}/get_network/${rippledUrl}`)
         .then((resp) => resp.data)
-        .then((data) => setNetworkName(data.network))
+        .then((data) =>
+          data.result && data.result === 'error'
+            ? setNetworkName(null)
```
Comment (ckniffen) [inline src/containers/App/App.jsx:60]: Have you tried putting the ternary inside `setNetworkName`?

File: src/containers/Network/index.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  const { params } = match
+  const { tab = 'nodes' } = params
```
Comment (ckniffen) [inline src/containers/Network/index.tsx:24]: Leverage `useParams`

**Review by ckniffen — APPROVED** [review]

File: public/locales/en-US/translations.json [hunk 5]
```diff
@@ -411,5 +411,8 @@
   "lp_tokens": "LP Tokens",
   "min_slot_price": "Minimum Slot Price",
   "max_slot_price": " Maximum Slot Price",
-  "auth_accounts": "Authorized Accounts"
+  "auth_accounts": "Authorized Accounts",
+  "network_cannot_crawled": "This network cannot be crawled",
```
Comment (mvadari) [inline public/locales/en-US/translations.json:415]:
```suggestion
  "network_cannot_be_crawled": "This network cannot be crawled",
```

**Review by mvadari — APPROVED** [review]

File: src/containers/Network/index.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    analytics(ANALYTIC_TYPES.pageview, {
+      title: 'network',
+      path: `/network/${tab || 'nodes'}`,
+    })
+  })
+
+  if (network === null) {
+    return (
+      <NoMatch
+        title="network_cannot_crawled"
```
Comment (ckniffen) [inline src/containers/Network/index.tsx:27]:
```suggestion
        title="network_cannot_be_crawled"
```

**Review by ckniffen — CHANGES_REQUESTED** [review]
Need to update usage of language key

**Comment by pdp2121** [comment]
> Need to update usage of language key

Thanks for noticing that. Made the update, pulled to make sure everything looks right.

**Pull request merged**